scripts/scil_tractogram_dps_math.py

```diff
@@ -0,0 +1,129 @@
+                        'key. Must be one of the following: [%(choices)s].\n'
+                        'The additional arguments required for each\n'
+                        'operation are specified under each group below.')
+    p.add_argument('dps_key', type=str,
+                   help='Where to find the data to be exported,\n'
+                        'in the tractogram.')
+
+    p.add_argument('--out_tractogram',
+                   help='Output tractogram (.trk). Required for any mutation.')
+
+    add_args = p.add_argument_group('Add operation',
+                                    'Requires the out_tractogram argument.')
+    add_args.add_argument('--add_dps_file',
+                          help='File containing the data to add to\n'
+                               'streamlines (.txt, .npy or .mat).')
+
+    _ = p.add_argument_group('Delete operation',
+                             'Requires the out_tractogram argument.')
+
+    export_args = p.add_argument_group('Export operation')
+    export_args.add_argument('--export_dps_file',
+                             help='File in which the extracted data will be\n'
+                                  'saved (.txt or .npy).')
+
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+    logging.getLogger().setLevel(logging.getLevelName(args.verbose))
+
+    check_tract_trk(parser, args.in_tractogram)
+    if args.out_tractogram:
+        check_tract_trk(parser, args.out_tractogram)
+
+    # I/O assertions
+    assert_inputs_exist(parser, args.in_tractogram, args.add_dps_file)
+    assert_outputs_exist(parser, args, [], optional=[args.export_dps_file,
+                                                     args.out_tractogram])
+
+    sft = load_tractogram(args.in_tractogram, 'same')
+
+    if args.operation == 'add':
+        # Make sure the user is not unwillingly overwritting dps
+        if (args.dps_key in sft.get_data_per_streamline_keys() and
+                not args.overwrite):
+            parser.error('"{}" already in data per streamline. Use -f to force'
+                         ' overwriting.'.format(args.dps_key))
+
+        # Load data and remove extraneous dimensions
+        data = np.squeeze(load_matrix_in_any_format(args.add_dps_file))
+
+        # Quick check as the built-in error from sft is not too explicit
+        if len(sft) != data.shape[0]:
+            raise ValueError('Data must have as many entries ({}) as there are'
+                             ' streamlines ({}).'.format(data.shape[0],
+                                                         len(sft)))
+
+        sft.data_per_streamline[args.dps_key] = data
+
+        save_tractogram(sft, args.out_tractogram)
+
+    if args.operation == 'delete':
+        del sft.data_per_streamline[args.dps_key]
+
+        save_tractogram(sft, args.out_tractogram)
+
+    if args.operation == 'export':
```

Comment:
you need to add some conditions to make sure export_dps_file has been provided or put a default output dps filename.